xapian-applications/omega/query.cc

```diff
@@ -651,25 +652,38 @@ html_strip(const string &str)
     return res;
 }
 
-// FIXME split list into hash or map and use that rather than linear lookup?
-static int word_in_list(const string& word, const string& list)
-{
-    string::size_type split = 0, split2;
-    int count = 0;
-    while ((split2 = list.find('\t', split)) != string::npos) {
-	if (word.size() == split2 - split) {
-	    if (memcmp(word.data(), list.data() + split, word.size()) == 0)
-		return count;
+static unordered_map<string, int> word_to_occurrence;
+
+class WordList {
+    string list;
+  public:
+    WordList() : list("") { }
```

Comment:
Thanks for sharing your excellent explanation. Lesson learnt: no need to explicitly initialise std::string (infact, it's better not to) :)